syntax = "proto3";



option java_multiple_files = true;

option java_package = "io.grpc.iiwaService";




service iiwaService {

  rpc askCartesianPosition (python_request) returns (robot_cartesian_position_reply) {}
  rpc askJointPosition (python_request) returns (robot_joint_position_reply) {}
  rpc move (python_cartesian_position_request) returns (robot_reply) {}
  rpc move_with_orientation (python_cartesian_pose_request) returns (robot_reply) {}

  rpc auto_down (force_condition) returns (robot_reply) {}

  rpc hold_position (python_cartesian_pose_request) returns (robot_reply) {}


}


message robot_reply{
  bool success = 1;
}
message python_request{}

message robot_cartesian_position_reply{
  double X = 1;
  double Y = 2;
  double Z = 3;
  double A = 4;
  double B = 5;
  double C = 6;
}

message robot_joint_position_reply{
  double A1 = 1;
  double A2 = 2;
  double A3 = 3;
  double A4 = 4;
  double A5 = 5;
  double A6 = 6;
  double A7 = 7;
}


message python_cartesian_position_request {

  double X = 1;
  double Y = 2;
  double Z = 3;

  double velocity = 4;

  bool mode = 5;
}



message python_cartesian_pose_request {

  double X = 1;
  double Y = 2;
  double Z = 3;

  double A = 4;
  double B = 5;
  double C = 6;

  double velocity = 7;

  bool mode = 8;
}

message force_condition {
  double force = 1;
  double 192.168.31.90 = 2;
  double velocity = 3;
}